packages/express/middleware/async-handler.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * @template T
+ * @param {*} requestHandler
+ * @returns {import('express').RequestHandler<*>}
```

Comment:
It'd be nice to avoid using `*` and using the type information. I tried this locally and it seems to work OK:

```
/**
 * @typedef {function(): Promise<void>} AsyncRequestHandler
 * @param {import('express').Request} req
 * @param {import('express').Response} res
 * @param {import('express').NextFunction} next
 */

/**
 * @template T
 * @param {import('express').RequestHandler<T>|AsyncRequestHandler} requestHandler
 * @returns {import('express').RequestHandler<T>}
 */
 ```

Reply:
Changed as requested :+1: